workers/unity/Assets/Fps/Scripts/SetupLogic/SimulatedPlayerCoordinatorWorkerConnector.cs

```diff
@@ -59,21 +60,26 @@ protected override async void Start()
 
         protected override IConnectionHandlerBuilder GetConnectionHandlerBuilder()
         {
-            var builder = new SpatialOSConnectionHandlerBuilder()
-                .SetConnectionParameters(CreateConnectionParameters(WorkerUtils.SimulatedPlayerCoordinator));
+            IConnectionFlow connectionFlow;
+            ConnectionParameters connectionParameters;
 
             var workerId = CreateNewWorkerId(WorkerUtils.SimulatedPlayerCoordinator);
 
             if (Application.isEditor)
             {
-                builder.SetConnectionFlow(new ReceptionistFlow(workerId));
+                connectionFlow = new ReceptionistFlow(workerId);
+                connectionParameters = CreateConnectionParameters(WorkerUtils.SimulatedPlayerCoordinator);
             }
             else
             {
-                builder.SetConnectionFlow(new ReceptionistFlow(workerId, new CommandLineConnectionFlowInitializer()));
+                connectionFlow = new ReceptionistFlow(workerId, new CommandLineConnectionFlowInitializer());
+                connectionParameters = CreateConnectionParameters(WorkerUtils.SimulatedPlayerCoordinator,
+                    new CommandLineConnectionParameterInitializer());
             }
 
-            return builder;
+            return new SpatialOSConnectionHandlerBuilder()
+                .SetConnectionFlow(connectionFlow)
+                .SetConnectionParameters(connectionParameters);
```

Comment:
This could be further refactored if you like: 


```
var workerId = CreateNewWorkerId(WorkerUtils.SimulatedPlayerCoordinator);
var initializer = Application.isEditor ? null : new CommandLineConnectionParameterInitializer();

return new SpatialOSConnectionHandlerBuilder()
	.SetConnectionFlow(new ReceptionistFlow(workerId, initializer)
	.SetConnectionParameters(CreateConnectionParameters(WorkerUtils.SimulatedPlayerCoordinator, initializer);

```

Reply:
big brain move - we don't need the isEditor check at all because the arguments would not be there anyway? the null would just save us the cost of attempting to get the values

Reply:
I do kind of prefer the explicit `isEditor` check. Otherwise its somewhat less clear

Reply:
in which case your code sample would actually have to be:

```csharp
var workerId = CreateNewWorkerId(WorkerUtils.SimulatedPlayerCoordinator);
var flowInitializer = Application.isEditor ? null: new CommandLineConnectionFlowInitializer();
var paramInitializer = Application.isEditor ? null : new CommandLineConnectionParameterInitializer();

return new SpatialOSConnectionHandlerBuilder()
	.SetConnectionFlow(new ReceptionistFlow(workerId, flowInitializer))
	.SetConnectionParameters(CreateConnectionParameters(WorkerUtils.SimulatedPlayerCoordinator, paramInitializer));
```

I prefer the current if (Application.isEditor) {} else {} block to that

Reply:
Sure 👍